pkg/logging/versioned_logger.go

```diff
@@ -0,0 +1,165 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logging
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+	"sync"
+
+	"k8s.io/klog/v2"
+	"kpt.dev/configsync/pkg/status"
+)
+
+const (
+	version             = "v1"
```

Comment:
I would prefer we NOT version the log format. 

Having a version makes it seem like an interface, but we are not planning to document it as an integration point or guarantee reverse compatibility. If customers write integrations with the logs, they might break after upgrades, but that's an acceptable compromise, because the primary reason we're using logs now instead of PubSub is to provide a temporary workaround without increasing KTLO costs.

Reply:
Corresponding change removed.